packages/http/src/validator/utils/__tests__/spec.spec.ts

```diff
@@ -1,16 +1,60 @@
 import { findOperationResponse } from '../spec';
 
 describe('findOperationResponse()', () => {
-  it('works', () => {
-    expect(
-      findOperationResponse(
-        [
-          { code: '2XX', contents: [], headers: [] },
-          { code: '20X', contents: [], headers: [] },
-          { code: '1XX', contents: [], headers: [] },
-        ],
-        200,
-      ),
-    ).toEqual({ code: '20X', contents: [], headers: [] });
+  describe('when response for given code exists', () => {
+    it('returns found response', () => {
+      expect(
+        findOperationResponse(
+          [
+            { code: '2XX', contents: [], headers: [] },
+            { code: '20X', contents: [], headers: [] },
+            { code: 'default', contents: [], headers: [] },
+            { code: '1XX', contents: [], headers: [] },
+          ],
+          200,
+        ),
+      ).toEqual({ code: '20X', contents: [], headers: [] });
+    });
+  });
+
+  describe('when response for given code does not exists but there is a default response', () => {
+    it('returns default response', () => {
+      expect(
+        findOperationResponse(
+          [
+            { code: '2XX', contents: [], headers: [] },
+            { code: 'default', contents: [], headers: [] },
+            { code: '1XX', contents: [], headers: [] },
+          ],
+          422,
+        ),
+      ).toEqual({ code: 'default', contents: [], headers: [] });
+    });
+  });
+
+  describe('when default response is specified with mixed case', () => {
+    it('returns default response', () => {
+      expect(
+        findOperationResponse(
+          [
+            { code: '2XX', contents: [], headers: [] },
+            { code: 'deFAULT', contents: [], headers: [] },
```

Comment:
This is interesting; I am not sure this should work. @philsturgeon shall the default response be case insensitive?

Reply:
I blindly assumed that someone can use write "default" using any case.

Reply:
The spec says it's either a number or `default`, named responses are not allowed. 

@philsturgeon will do the call here, but I think we should not test/support this scenario.

Reply:
It is not "named response". It is "default" string being case insensitive vs sensitive.

Just in case some folk will make:
```
...
responses:
  200: ...
  3XX: ...
  DEFAULT:
...
```

Reply:
Yeah I get it; my point is that if the response is not a number and it's not `default` then it's a named response (as in a regular string).

I still think we should not support these scenarios at all as they're invalid OAS documents and spectral will likely refuse them anyway.

In any case, @philsturgeon (he's bombarded with mentions now :trollface:) will answer the question for us.

Reply:
It’s difficult to maintain a difficult balance between “only supporting what the spec supports” and “supporting what people do” but yeah the options here are “it’s an integer as a string because YAML” or its “default“ but we can ignore everything else. Let’s not waste our time on conventions like 3xx == 300 or anything else.

Reply:
Noted. @StefanDywersant can you make sure we only look for `default` and not for any other of its combinations?

Reply:
Done 👍